net/http/http_cache_transaction.cc

```diff
@@ -4071,6 +4071,21 @@ bool HttpCache::Transaction::ShouldDisableCaching(
     return true;
   }
 
+#if BUILDFLAG(IS_STARBOARD)
+  // Similar to the mime-type allow list in Cobalt 25, except images are not
+  // cacheable. Consider making this configurable perhaps with command-line
+  // switches.
+  std::string mime_type;
+  headers.GetMimeType(&mime_type);
+  if (mime_type == "text/javascript" || mime_type == "text/html") {
```

Comment:
I think you need to include `application/javascript` here as well, just to be sure.

Going off the pared down list here https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/25.lts.1+:net/http/http_cache_transaction.cc;l=78?q=kMimeTypesCacheAllowlist

Other than that, it looks reasonable and replicates the C25 logic. Image formats are excluded as they were anyway excluded with quota = 0.